### ISSUE TITLE
Menu stack off by 1 when opening submenu with enter or space key

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-core#2015]
# Bug Report

## Description
I noticed this issue when testing via keyboard. Opening the submenu by focusing on the menu item that triggers open the submenu and pressing the `Enter` or `Space` key causes the menu stack to increase by 2 instead of 1.

If you click on the menu item that triggers open the submenu or when focused on the menu item that triggers open the submenu and press right arrow key, only 1 item is added to the array `stack`.

This results in additional keyboard events needed to close the submenu. The user has to hit enter or space on the back button twice to reduce that stack to its initial state.

## Steps to Reproduce

1. Navigate to https://engineering.cerner.com/terra-core/#/raw/tests/terra-menu/menu/menu/sub-menu
1. Click on the button on the page
1. Open the react dev tools and select the menu component in the DOM so you can view it's state. The `stack` array in state should have just 1 item in it.
1. Press `tab` key, then press `down arrow` key
1. On this step, press either `Enter` or `Space` key. 
1. 2 items are added to the array `stack` totaling `3` items in the `stack`.

To close the submenu with enter, you have to press `tab` and then `enter` and then `tab` and then `enter` again to reduce the `stack` back down to 1 item.

Video of steps from above.
[menu-stack-issue.mp4.zip](https://github.com/cerner/terra-core/files/2539951/menu-stack-issue.mp4.zip)

## Expected Behavior
Pressing enter or space on the menu item that triggers open the submenu should only increase the stack by one.

## Possible Solution
Look into event handlers and see how we are getting this error and refactor it.

## Environment

* Component Name and Version: terra-menu, terra-collapsible-menu-view
* Browser Name and Version: Tested in Mobile Safari 12 w/ VoiceOver on iOS and Chrome
* Operating System and version (desktop or mobile): Mac, iOS 12
